### PR TITLE
Fix: Register application/applicationInstance service return null

### PR DESCRIPTION
**Pull request by liuhaoyang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

___
### Bug fix
- Bug description.
Call `registerApplicationInstance` service return empty result
- How to fix?
Polling the asynchronous return value until get the result

### CODE REVIEW

File: src/SkyWalking.Core/Remote/GrpcApplicationService.cs [hunk 1]
```diff
@@ -45,96 +45,120 @@ public class GrpcApplicationService : TimerService
...
+                    AgentUUID = agentUUID,
+                    RegisterTime = registerTime,
+                    Osinfo = osInfo
+                };
+
+                var retry = 0;
+                var applicationInstanceId = 0;
+                while (retry++ < 5 && DictionaryUtil.IsNull(applicationInstanceId))
+                {  
+                    var applicationInstanceMapping =await instanceDiscoveryService.registerInstanceAsync(applicationInstance);
```
Comment (yuleyule66) [inline src/SkyWalking.Core/Remote/GrpcApplicationService.cs]: ~~`registerInstanceAsync()` Whether to follow the C# code specification?~~
I find this is due to code generator code, so we can ignore that.

File: src/SkyWalking.Core/Remote/GrpcApplicationService.cs [hunk 2]
```diff
@@ -45,96 +45,120 @@ public class GrpcApplicationService : TimerService
...
+                {
+                    ApplicationId = RemoteDownstreamConfig.Agent.ApplicationId,
+                    AgentUUID = agentUUID,
+                    RegisterTime = registerTime,
+                    Osinfo = osInfo
+                };
+
+                var retry = 0;
+                var applicationInstanceId = 0;
+                while (retry++ < 5 && DictionaryUtil.IsNull(applicationInstanceId))
```
Comment (yuleyule66) [inline src/SkyWalking.Core/Remote/GrpcApplicationService.cs]: Do we use `try catch` to ensure that `retry++<5` can execute normally.

File: src/SkyWalking.Core/Remote/GrpcConnection.cs [hunk 3]
```diff
@@ -52,8 +52,7 @@ public GrpcConnection(string server)
             _state = GrpcConnectionState.Connecting;
             try
             {
-                // default timeout = 5s
-                var deadLine = DateTime.UtcNow.AddSeconds(5);
+                var deadLine = DateTime.UtcNow.AddSeconds(timeout);
                 await _internalChannel.ConnectAsync(deadLine);
```
[the remark this answers is not in the document]
Comment (liuhaoyang) [reply src/SkyWalking.Core/Remote/GrpcConnection.cs]: Grpc library only provides asynchronous `ConnectAsync` method

File: src/SkyWalking.Core/Remote/GrpcConnection.cs [hunk 4]
```diff
@@ -43,7 +43,7 @@ public GrpcConnection(string server)
             _internalChannel = new Channel(server, ChannelCredentials.Insecure);
         }
       
-        public async Task<bool> ConnectAsync()
+        public async Task<bool> ConnectAsync(long timeout)
```
[the remark this answers is not in the document]
Comment (liuhaoyang) [reply src/SkyWalking.Core/Remote/GrpcConnection.cs]: Agree !

**Pull request merged**